rust/src/wrapper/hci.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! HCI
+
+use itertools::Itertools as _;
+use pyo3::{exceptions::PyException, intern, types::PyModule, PyErr, PyObject, PyResult, Python};
+
+/// A Bluetooth address
+pub struct Address(pub(crate) PyObject);
+
+impl Address {
+    /// The type of address
+    pub fn address_type(&self) -> PyResult<AddressType> {
+        Python::with_gil(|py| {
+            let addr_type = self
+                .0
+                .getattr(py, intern!(py, "address_type"))?
+                .extract::<u32>(py)?;
+
+            let module = PyModule::import(py, intern!(py, "bumble.hci"))?;
+            let klass = module.getattr(intern!(py, "Address"))?;
+
+            if addr_type
```

Comment:
nit: use a `match` ?

Reply:
match didn't work because the things being matched against aren't literals